saleor/graphql/shop/resolvers.py

```diff
@@ -0,0 +1,44 @@
+from ...shipping.models import ShippingMethod, ShippingMethodChannelListing
+from ..channel import ChannelContext
+
+
+def resolve_available_shipping_methods(info, channel_slug: str, address):
+    available = ShippingMethod.objects.filter(
+        channel_listings__channel__slug=channel_slug
+    )
+    if address and address.country:
+        available = available.filter(
+            shipping_zone__countries__contains=address.country,
+        )
+
+    if available is None:
+        return []
+    manager = info.context.plugins
+    display_gross = info.context.site.settings.display_gross_prices
+    shipping_mapping = get_shipping_method_to_shipping_price_mapping(
+        available, channel_slug
+    )
+    for shipping_method in available:
+        shipping_price = shipping_mapping[shipping_method.pk]
+        taxed_price = manager.apply_taxes_to_shipping(shipping_price, address)
```

Comment:
Have you checked this with avalara? I am afraid that current implementation assumes that it will get the full address data

Reply:
I'm working on a fix for `Vatlayer` (`Avalara` doesn't have `apply_taxes_to_shipping` method).